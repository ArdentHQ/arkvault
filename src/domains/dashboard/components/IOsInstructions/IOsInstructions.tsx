import React from "react";
import { useTranslation } from "react-i18next";

import { Button } from "@/app/components/Button";
import { Image } from "@/app/components/Image";

export const IOsInstructions = ({ onClose }: { onClose: () => void }) => {
	const { t } = useTranslation();

	return (
		<div className="mt-8" data-testid="IOsInstructions">
			<div className="flex flex-col items-start">
				<div className="mb-4 font-semibold text-theme-secondary-text">{t("COMMON.INSTALL_ARKVAULT_STEP1")}</div>
				<Image name="IOsInstructionsStep1" className="mb-8 h-auto w-full" />

				<div className="mb-4 font-semibold text-theme-secondary-text">{t("COMMON.INSTALL_ARKVAULT_STEP2")}</div>
				<Image name="IOsInstructionsStep2" className="mb-8 h-auto w-full" />

				<div className="mb-4 font-semibold text-theme-secondary-text">{t("COMMON.INSTALL_ARKVAULT_STEP3")}</div>
				<Image name="IOsInstructionsStep3" className="h-auto w-full" />
			</div>

			<div className="mt-8 flex items-end justify-end">
				<Button type="submit" onClick={onClose} variant="secondary" data-testid="IOsInstructions__close-button">
					{t("COMMON.CLOSE")}
				</Button>
			</div>
		</div>
	);
};
