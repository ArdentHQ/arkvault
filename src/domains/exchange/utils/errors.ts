interface ErrorWithResponse extends Error {
	response: any;
}

const isInvalidAddressError = (error: ErrorWithResponse) => {
	try {
		return error.response.json().error.message === "Invalid Address";
	} catch {
		return false;
	}
};

const isInvalidRefundAddressError = (error: ErrorWithResponse) => {
	try {
		return error.response.json().error.message === "Invalid Refund Address";
	} catch {
		return false;
	}
};

const isUnavailablePairError = (error: ErrorWithResponse) => {
	try {
		return error.response.json().error.message === "Unavailable Pair";
	} catch {
		return false;
	}
};

export { isInvalidAddressError, isInvalidRefundAddressError, isUnavailablePairError };
