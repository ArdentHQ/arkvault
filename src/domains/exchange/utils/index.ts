export * from "./assertions";
export * from "./errors";
