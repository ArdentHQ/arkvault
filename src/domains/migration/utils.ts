import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
