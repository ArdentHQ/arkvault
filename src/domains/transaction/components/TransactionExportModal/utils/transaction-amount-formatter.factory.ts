import { BigNumber } from "@ardenthq/sdk-helpers";
import { DTO, Helpers } from "@ardenthq/sdk-profiles";

const formatAmount = (amount: number, transaction: DTO.ExtendedConfirmedTransactionData) =>
	Helpers.Currency.format(amount, transaction.wallet().exchangeCurrency(), {
		withTicker: false,
	});

const multiPaymentAmount = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isReceived()) {
		let totalReceived = BigNumber.make(transaction.amount());

		for (const recipient of transaction.recipients()) {
			if (recipient.address === transaction.wallet().address()) {
				totalReceived.minus(recipient.amount);
			}
		}

		return formatAmount(totalReceived.toNumber(), transaction);
	}

	return formatAmount(transaction.amount(), transaction);
};

const transactionAmount = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isMultiPayment()) {
		return multiPaymentAmount(transaction);
	}

	return formatAmount(transaction.amount(), transaction);
};

const transactionTotal = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isMultiPayment()) {
		return BigNumber.make(multiPaymentAmount(transaction)).plus(transaction.fee());
	}

	return formatAmount(transaction.total(), transaction);
};

export const AmountFormatter = (transaction: DTO.ExtendedConfirmedTransactionData) => ({
	amount: () => transactionAmount(transaction),
	fee: () => formatAmount(transaction.fee(), transaction),
	total: () => transactionTotal(transaction),
});
