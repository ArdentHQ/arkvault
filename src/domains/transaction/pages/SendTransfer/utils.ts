export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
