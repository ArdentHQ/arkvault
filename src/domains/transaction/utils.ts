import { Services } from "@ardenthq/sdk";
import { RecipientItem } from "@/domains/transaction/components/RecipientList/RecipientList.contracts";

export const isNoDeviceError = (error: any) => {
	if (!error) {
		return false;
	}

	if (String(error).includes("no device found")) {
		return true;
	}

	return false;
};

export const isRejectionError = (error: any) => String(error).includes("Condition of use not satisfied");

export const handleBroadcastError = ({ errors }: Services.BroadcastResponse) => {
	const allErrors = Object.values(errors);

	if (allErrors.length === 0) {
		return;
	}

	throw new Error(allErrors[0]);
};

export const getTransferType = ({ recipients }: { recipients: RecipientItem[] }): "multiPayment" | "transfer" => {
	const isMultiPayment = recipients.length > 1;
	return isMultiPayment ? "multiPayment" : "transfer";
};

export const withAbortPromise =
	(signal?: AbortSignal, callback?: () => void) =>
	<T>(promise: Promise<T>) =>
		new Promise<T>((resolve, reject) => {
			if (signal) {
				signal.addEventListener("abort", () => {
					callback?.();
					reject("ERR_ABORT");
				});
			}

			return promise.then(resolve).catch(reject);
		});
