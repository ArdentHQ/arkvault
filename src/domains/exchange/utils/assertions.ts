import { AssertionError } from "assert";
import { Contracts } from "@ardenthq/sdk-profiles";

import { CurrencyData } from "@/domains/exchange/exchange.contracts";
import { ExchangeService } from "@/domains/exchange/services/exchange.service";

export function assertExchangeService(exchangeService?: ExchangeService): asserts exchangeService is ExchangeService {
	if (!(exchangeService instanceof ExchangeService)) {
		throw new AssertionError({
			message: `Expected 'exchangeService' to be ExchangeService, but received ${exchangeService}`,
		});
	}
}

export function assertExchangeTransaction(
	exchangeTransaction?: Contracts.IExchangeTransaction,
): asserts exchangeTransaction is Contracts.IExchangeTransaction {
	if (typeof exchangeTransaction?.orderId !== "function") {
		throw new AssertionError({
			message: `Expected 'exchangeTransaction' to be Contracts.IExchangeTransaction, but received ${exchangeTransaction}`,
		});
	}
}

export function assertCurrency(currencyData?: CurrencyData): asserts currencyData is CurrencyData {
	if (!currencyData || typeof currencyData !== "object" || !("coin" in currencyData && "name" in currencyData)) {
		throw new AssertionError({
			message: `Expected 'currencyData' to be CurrencyData, but received ${currencyData}`,
		});
	}
}
