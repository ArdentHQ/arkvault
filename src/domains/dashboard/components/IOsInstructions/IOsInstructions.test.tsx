import userEvent from "@testing-library/user-event";
import React from "react";

import { render, screen } from "@/utils/testing-library";

import { IOsInstructions } from "./IOsInstructions";

describe("IOsInstructions", () => {
	it("should render ios instructions", () => {
		const { container } = render(<IOsInstructions onClose={vi.fn()} />);

		expect(container).toMatchSnapshot();
	});

	it("should handle close", async () => {
		const onClose = vi.fn();
		render(<IOsInstructions onClose={onClose} />);

		await userEvent.click(screen.getByTestId("IOsInstructions__close-button"));

		expect(onClose).toHaveBeenCalledWith(expect.anything());
	});
});
