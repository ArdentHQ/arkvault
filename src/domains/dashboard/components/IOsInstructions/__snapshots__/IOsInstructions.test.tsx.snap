// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IOsInstructions should render ios instructions 1`] = `
<div>
  <div
    class="mt-8"
    data-testid="IOsInstructions"
  >
    <div
      class="flex flex-col items-start"
    >
      <div
        class="mb-4 font-semibold text-theme-secondary-text"
      >
        1. Press the share button
      </div>
      <svg
        class="mb-8 h-auto w-full"
      >
        ios-instructions-step-1-light.svg
      </svg>
      <div
        class="mb-4 font-semibold text-theme-secondary-text"
      >
        2. Select "Add to Home Screen"
      </div>
      <svg
        class="mb-8 h-auto w-full"
      >
        ios-instructions-step-2-light.svg
      </svg>
      <div
        class="mb-4 font-semibold text-theme-secondary-text"
      >
        3. Press the "Add" button
      </div>
      <svg
        class="h-auto w-full"
      >
        ios-instructions-step-3-light.svg
      </svg>
    </div>
    <div
      class="mt-8 flex items-end justify-end"
    >
      <button
        class="css-1jhy6v9"
        data-testid="IOsInstructions__close-button"
        type="submit"
      >
        <div
          class="flex items-center space-x-2"
        >
          Close
        </div>
      </button>
    </div>
  </div>
</div>
`;
